### ISSUE TITLE
possible to Duplicate / one-way sync databases?

### ISSUE THREAD

**Issue by abismail** [JumpMind/metl#732]
Hi guys,

We were trying out SymmetricDS to create a one-way sync from a slave db to a duplicate database that'll eventually be used for reporting. That took forever so we bailed on that.

But now I'm stuck create a single model for each table in a 150 table big database and that task doesn't exactly tickle my funny bone. There are other ways I can do this, but I've tried a couple and I'm not seeing any light at the end of the tunnel. Here are the things I've tried / wanted to try:

- creating a model for one specific table, getting the data from a source db using an RDBMS reader and writing it to a target db using an RDBMS writer. This was successful using a single model for input and output on the RDBMS reader and input on the RDBMS writer. however I have to create multiple)

- Importing all tables from a source db into a model. I successfully imported the structure of all tables in a source database into a model. >> I couldn't run a select on all tables to read it into this model, once I import more than one entity / table into a model, data can no longer be read into it.

Which brings me to my question: Is there a component / set of components / method (any suggestion / theory works) to read all data including tables structure from a source database and duplicate that in a single destination?

**Comment by gwilmer** [comment]
Abismail,

First, I would be interested to know why attempting to setup SymmetricDS took forever.  What do you mean by that?  Took forever to set up?  Took forever to sync the databases once you had it set up?  SymmetricDS probably is the way you want to go for this if you want real time sync between the two databases.

Second, on the Metl side it sounds like you created a model that contains every table in your source database (hopefully by importing the model from the datasource itself).  Once you do that you should be able to create an rdbms reader and rdbms writer (as long as the table structure matches between the two databases).  In your rdbms reader, your sql script would just be something like this:

select * From table1;
select * From table2;
select * From table3;
etc.

If you wanted to transform the data, you would need to add additional components.  In addition, each time you ran this flow, it would do a full file refresh of your tables to the second node (not very efficient).  That's where SymmetricDS would be a much better fit as it would do change data capture.

Hope this helps.

**Comment by abismail** [comment]
There's not many tutorials that can help me sync MY tables across symmetricDS nodes, the symmetric tables themselves synced fine, but I was hitting roadblock after road block and at the end there I needed a way to run " set sql_mode=''; " before the sync and I'm sure that I would've progressed easier, but frankly using symmetricDS for a one way sync like that would be like frying a whale at a dinner for two. (way overkill)

So anyways:

I retried and it worked... but now I'm having issues when I have zero valued date timestamps. I converted the data types in the input / output models to varchar, but it's still trying to load it into a Java timestamp wth?
